File: Python/Program/Dosen.py
# Import File:
from Human import Human

# Kelas Dosen (Child) - Human (Parent)
class Dosen(Human):
    # Atribut Mahasiswa:
    __nip = ""
    __fakultas = ""
    __pend_terakhir = ""
    __keahlian = ""
    # Atribut komponen kelas course.
    __course = []

    # Konstruktor
    def __init__(self, nik, nama, jenis_kelamin, nip, fakultas, pend_terakhir, keahlian, course):
        Human.__init__(self, nik, nama, jenis_kelamin)
        self.__nip = nip
        self.__fakultas = fakultas
        self.__pend_terakhir = pend_terakhir
        self.__keahlian = keahlian
        self.__course = course.copy()

    # Setter
    def set_nip(self, nip):
        self.__nip = nip

    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas

    def set_pend_terakhir(self, pend_terakhir):
        self.__pend_terakhir = pend_terakhir

    def set_keahlian(self, keahlian):
        self.__keahlian = keahlian

    def set_course(self, course):
        self.__course = course.copy()

    # Getter
    def get_nip(self):
        return self.__nip

    def get_fakultas(self):
        return self.__fakultas
    
    def get_pend_terakhir(self):
        return self.__pend_terakhir
    
    def get_keahlian(self):
        return self.__keahlian
    
    def get_course(self):
        return self.__course

File: CPP/Program/Dosen.cpp
// Import file:
#include "header.hh"

// Empty konstruktor.
Dosen::Dosen() : Human()
{
    this->nip = "";
    this->fakultas = "";
    this->pend_terakhir = "";
    this->keahlian = "";
    this->course = {};
}

// Konstruktor with params.
Dosen::Dosen(string nik, string nama, string jenis_kelamin, string nip, string fakultas, string pend_terakhir, string keahlian, vector<Course> course) : Human(nik, nama, jenis_kelamin)
{
    this->nip = nip;
    this->fakultas = fakultas;
    this->pend_terakhir = pend_terakhir;
    this->keahlian = keahlian;
    this->course = course;
}

// Setter
void Dosen::set_nip(string nip)
{
    this->nip = nip;
}
void Dosen::set_fakultas(string fakultas)
{
    this->fakultas = fakultas;
}
void Dosen::set_pend_terakhir(string pend_terakhir)
{
    this->pend_terakhir = pend_terakhir;
}
void Dosen::set_keahlian(string keahlian)
{
    this->keahlian = keahlian;
}
void Dosen::set_course(vector<Course> course)
{
    this->course = course;
}

// Getter
string Dosen::get_nip()
{
    return this->nip;
}
string Dosen::get_fakultas()
{
    return this->fakultas;
}
string Dosen::get_pend_terakhir()
{
    return this->pend_terakhir;
}
string Dosen::get_keahlian()
{
    return this->keahlian;
}
vector<Course> Dosen::get_course()
{
    return this->course;
}

Dosen::~Dosen()
{
    
}